packages/@aws-cdk/aws-appsync/lib/caching-config.ts

```diff
@@ -0,0 +1,22 @@
+import { Duration } from '@aws-cdk/core';
+
+/**
+ * CachingConfig for AppSync resolvers
+ */
+export interface CachingConfig {
+  /**
+   * The caching keys for a resolver that has caching enabled.
+   * Valid values are entries from the $context.arguments, $context.source, and $context.identity maps.
```

Comment:
Can you please also add validation for this constraint?

Reply:
Will do!

Reply:
Alright - added validation for caching keys - added some constants for $context.source, $context.arguments and $context.identity to help validate.

Reply:
Thanks! One last thing: Before validating you should check whether the value is [encoded as a token][token], using the `Token.isUnresolved()` static method.

[token]: https://docs.aws.amazon.com/cdk/latest/guide/tokens.html#tokens_encoding

Reply:
Ah cool - I'll make this change!

Reply:
Alright sweet - done! Checked to see if the caching key is resolved first before validating.